watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/metab/multi_sweep.py

```diff
@@ -0,0 +1,169 @@
+            3.376,
+            rep_factor_upper_lim,
+            nx,
+        )
+
+    elif case_num == 3:
+        # Hydrogen METAB HRT
+        sweep_params["hydrogen_hrt"] = LinearSample(
+            m.fs.metab_hydrogen.hydraulic_retention_time, 0.75, 24, nx
+        )
+
+    elif case_num == 4:
+        # Methane METAB HRT
+        sweep_params["methane_hrt"] = LinearSample(
+            m.fs.metab_methane.hydraulic_retention_time, 47.25, 360, nx
+        )
+
+    elif case_num == 5:
+        # Hydrogen Conversion Rate: sweep from 0.06 to 0.6 L H2/g-COD removed
+        sweep_params["hydrogen_conversion_rate"] = LinearSample(
+            m.fs.metab_hydrogen.generation_ratio["cod_to_hydrogen", "hydrogen"],
+            5.03e-3,
+            5e-2,
+            nx,
+        )
+
+    elif case_num == 6:
+        # Methane Conversion Rate: sweep from ~ 0.1 to 0.3 L H2/g-COD removed
+        sweep_params["methane_conversion_rate"] = LinearSample(
+            m.fs.metab_methane.generation_ratio["cod_to_methane", "methane"],
+            6.68e-2,
+            2e-1,
+            nx,
+        )
+    elif case_num == 7:
+        # Recovered resource selling price:
+        sweep_params["Hydrogen selling price"] = LinearSample(
+            m.fs.costing.hydrogen_product_cost, -2, -10, nx
+        )
+        sweep_params["Methane selling price"] = LinearSample(
+            m.fs.costing.methane_product_cost, -0.305, -1, nx
+        )
+
+    else:
+        raise ValueError("case_num = %d not recognized." % (case_num))
+
+    output_filename = "sensitivity_" + str(case_num) + ".csv"
+
+    global_results = parameter_sweep(
+        m,
+        sweep_params,
+        outputs,
+        csv_results_file_name=output_filename,
+        optimize_function=opt_function,
+        optimize_kwargs=optimize_kwargs,
+        # debugging_data_dir=os.path.split(output_filename)[0] + '/local',
+        interpolate_nan_outputs=interp_nan_outputs,
+    )
+
+    return global_results, sweep_params
+
+
+if __name__ == "__main__":
+
+    # Start MPI communicator
+    comm, rank, num_procs = _init_mpi()
+
+    # Get the case number to run
+    try:
+        case_num = int(sys.argv[1])
+    except:
+        # Default to running case 1
+        case_num = 1
+
+    # Get the default number of discretization points
+    try:
+        nx = int(sys.argv[2])
+    except:
+        nx = 11
+
+    tic = time.time()
+    global_results, sweep_params = run_analysis(case_num, nx)
+    print(global_results)
+    toc = time.time()
+
+    if rank == 0:
+        total_samples = 1
+
+        for k, v in sweep_params.items():
+            total_samples *= v.num_samples
+
+        print("Finished case_num = %d." % (case_num))
+        print(
+            "Processed %d swept parameters comprising %d total points."
+            % (len(sweep_params), total_samples)
+        )
+        print("Elapsed time = %.1f s." % (toc - tic))
```

Comment:
It wouldn't be too much more work to get coverage on these lines by creating a `main` function.

Reply:
Ok feel free to make the change